File: use-cases/customer-growth-job-service/e2e/deploy/python-sqlalchemy-service/app/controller/customer_growth_job_controller.py
from __future__ import annotations

from app.dto.customer_growth_job_dto import (
    CustomerGrowthJobDTO,
    CustomerGrowthJobSearchRequest,
    CreateCustomerGrowthJobRequest,
    UpdateCustomerGrowthJobRequest,
)
from app.service.customer_growth_job_service import CustomerGrowthJobService


class CustomerGrowthJobController:
    def __init__(self, service: CustomerGrowthJobService) -> None:
        self._service = service

    def create_job(self, request: CreateCustomerGrowthJobRequest) -> CustomerGrowthJobDTO:
        return self._service.create_job(request)

    def get_job(self, job_id: int) -> CustomerGrowthJobDTO | None:
        return self._service.get_job(job_id)

    def update_job(self, job_id: int, request: UpdateCustomerGrowthJobRequest) -> CustomerGrowthJobDTO:
        return self._service.update_job(job_id, request)

    def delete_job(self, job_id: int) -> None:
        self._service.delete_job(job_id)

    def list_visible_jobs(
        self,
        request: CustomerGrowthJobSearchRequest,
    ) -> list[CustomerGrowthJobDTO]:
        return self._service.list_visible_jobs(request)

File: use-cases/customer-growth-job-service/e2e/deploy/golang-sqlx-service/pkg/controller/customer_growth_job_controller.go
package controller

import (
	"context"

	"authguard/use-cases/customer-growth-job-service/e2e/deploy/golang-sqlx-service/pkg/dto"
	"authguard/use-cases/customer-growth-job-service/e2e/deploy/golang-sqlx-service/pkg/service"
)

type CustomerGrowthJobController struct {
	service *service.CustomerGrowthJobService
}

func NewCustomerGrowthJobController(service *service.CustomerGrowthJobService) *CustomerGrowthJobController {
	return &CustomerGrowthJobController{service: service}
}

func (c *CustomerGrowthJobController) CreateJob(ctx context.Context, request dto.CreateCustomerGrowthJobRequest) (dto.CustomerGrowthJobDTO, error) {
	return c.service.CreateJob(ctx, request)
}

func (c *CustomerGrowthJobController) GetJob(ctx context.Context, id int64) (dto.CustomerGrowthJobDTO, bool, error) {
	return c.service.GetJob(ctx, id)
}

func (c *CustomerGrowthJobController) UpdateJob(ctx context.Context, id int64, request dto.UpdateCustomerGrowthJobRequest) (dto.CustomerGrowthJobDTO, error) {
	return c.service.UpdateJob(ctx, id, request)
}

func (c *CustomerGrowthJobController) DeleteJob(ctx context.Context, id int64) error {
	return c.service.DeleteJob(ctx, id)
}

func (c *CustomerGrowthJobController) ListVisibleJobs(ctx context.Context, request dto.CustomerGrowthJobSearchRequest) ([]dto.CustomerGrowthJobDTO, error) {
	return c.service.ListVisibleJobs(ctx, request)
}
